<launch>
    <arg name="model" default="$(find orientable_cartesian)/urdf/orientable_cartesian.xacro"/>
    <arg name="gui" default="false"/>
    <arg name="rvizconfig" default="$(find orientable_cartesian)/rviz/urdf.rviz" />

    <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)"/>
    <param name="use_gui" value="$(arg gui)"/>
    
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
        <param name="use_gui" value="$(arg gui)"/>
        <rosparam param="source_list">["joints_set"]</rosparam>
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
</launch>
